oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java

```diff
@@ -202,6 +211,14 @@ private static void mapIndexRules(ElasticIndexDefinition indexDefinition, XConte
                 }
             }
             mappingBuilder.endObject();
+
+            if (useInSimilarity) {
+                mappingBuilder.startObject(FieldNames.createSimilarityFieldName(name));
+                mappingBuilder.field("type", ES_DENSE_VECTOR_TYPE);
+                mappingBuilder.field(ES_DENSE_VECTOR_DIM_PROP, Integer.parseInt(System.getProperty(PROP_ES_SIMILARITY_SEARCH_DENSE_VECTOR_SIZE,
```

Comment:
Hm, this is using System.getProperty... I think it would be better if this is configurable in the index definition itself... Would that be possible?

Reply:
I think vector representation would be obtained from an external service and it would provide same length of vectors for all images. So duplicating this information for each index doesn't make sense currently. If such a need arises in future, it can be done without breaking backwards compatibility. WDYT?

Reply:
In my view, there's likely only one index that uses image similarity... Yes I think it makes sense to specify this in the index definition. Sure we can do it later if we are in a rush, but I would probably do it now...